### ISSUE TITLE
Pneumonia Detection using CNN(92.6% Accuracy) - 3 (learning_rate 초기 설정과 조절 / 데이터 로그 변환 / loss 함수 설정)

### ISSUE THREAD

**Issue by UGeunJi** [UGeunJi/Deep_Learning_Project#6]
#### 콜백 함수 ReduceLROnPlateau - patience = 2 -> 5, factor = 0.3 -> 0.2

![image](https://user-images.githubusercontent.com/84713532/211948152-4d64388e-3a35-43c0-8e35-db55ce06db1e.png)

성능이 낮아졌다. patience를 높였으니 epoch도 늘려봐야겠다.

---

#### 콜백 함수 ReduceLROnPlateau - patience = 2 -> 5, factor = 0.3 -> 0.2 + epoch = 12 -> 25

![image](https://user-images.githubusercontent.com/84713532/211948915-704fe379-e7da-43e7-955c-8fe8cdb00e04.png)

성능이 향상되었다. 92%에 매우 근접하다. epoch를 더 늘려봐야겠다.

---

#### 콜백 함수 ReduceLROnPlateau - patience = 2 -> 5, factor = 0.3 -> 0.2 + epoch = 25 -> 35

- 모델 정확도
![image](https://user-images.githubusercontent.com/84713532/211949889-f192f42c-0a4f-4334-967a-1e11971a62fe.png)

- 정밀도, 정확도, 재현율 등등
![image](https://user-images.githubusercontent.com/84713532/211949929-170976ce-f421-4130-8722-782fd522a078.png)

모델의 정확도는 91%로 이전보다 더 향상되진 않았지만, 높은 정확도다. 주목할 부분은 폐렴 클래스의 재현률이 굉장히 높다는 것이다.
min_lr을 더 낮춰서 다시 실행시켜 봐야겠다.

---

#### 나머지 이전과 동일 / min_lr=0.000001 -> 0.0000001

![image](https://user-images.githubusercontent.com/84713532/211951646-d058ffc2-e62b-44a7-aa36-2ada81dcfe53.png)

92%를 넘겼다. min_lr을 더 낮추고 patience를 4로 낮춰보자

---

#### 나머지 이전과 동일 / min_lr=0.0000001 -> 0.00000001, patience = 5 -> 4

![image](https://user-images.githubusercontent.com/84713532/211952806-58bcf45a-d904-410b-b46c-7e1927db929a.png)

나쁘지 않은 정확도가 나왔다. 이 방법으론 이제 큰 변화는 없는 것 같으니 다른 파라미터를 조정해 봐야겠다.

---

#### Binary Cross Entropy -> Categorical Cross Entropy

![image](https://user-images.githubusercontent.com/84713532/211953901-194f08a6-4111-4dbc-8617-df0c994068e6.png)

오류 뜬다. 적절치 않은 데이터에 사용해서 그런 것 같다. sparse_categorical_crossentropy 이 함수는 사용할 수 있다고 했으니 사용해 보겠다.

---

#### Binary Cross Entropy -> Sparse Categorical Cross Entropy

![image](https://user-images.githubusercontent.com/84713532/211954983-5212552a-d644-4380-9ab9-fafde48d41be.png)

다중 분류 문제에 쓰이는 함수를 쓰니 이런 결과가 나왔다. 문제에 따라 적절한 손실 함수가 쓰인다는 걸 알게 되었다.

---

#### Sparse Categorical Cross Entropy -> mse

![image](https://user-images.githubusercontent.com/84713532/211960820-b7d22312-9f45-4e8b-b770-280d4e76c75e.png)

매우 낮은 정확도가 나왔다. 이진 분류에는 이진 분류에 맞는 손실 함수를 사용해야 한다.

---

### ※ 로그 변환

![image](https://user-images.githubusercontent.com/84713532/211969605-f04b998d-1e8b-42ad-b3e3-71375ff0379f.png)

데이터 로그 변환은 어떤 이유에서 인지 정확도가 개선은 물론 달라지지도 않았다.

![image](https://user-images.githubusercontent.com/84713532/211969710-14315087-4aed-4de4-b348-ae6b2cba1ea1.png)

로그 변환은 사용하지 말아야겠다.

---

## 결론

learning rate 조절에서 미세하게 정확도가 향상된 것을 확인할 수 있었다. 다른 방법들에서는 개선된 결과를 볼 수 없었다.
